### PR TITLE
Add "jenkins" extras

### LINKED ISSUE #8228: Add "jenkins" extras requirements
https://github.com/chainer/chainer-test/blob/b1b40421a2d7c73b86590575b19b82c801a12778/run_combination_test.py#L54-L66

Move these requirements defined in [chainer-test](https://github.com/chainer/chainer-test) (there are some more files which has the same definition) to a new extras "jenkins" in `setup.py`.

### PR DESCRIPTION
**Pull request by niboshi**
See #8228

#8228 will be completed by chainer/chainer-test#530

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,6 +59,16 @@
         # TODO(niboshi): Consider upgrading pytest to >=3.6
         'pytest-timeout<1.3.0',
     ],
+    'jenkins': [
+        '-r test',
+        # pytest-timeout>=1.3.0 requires pytest>=3.6.
+        # TODO(niboshi): Consider upgrading pytest to >=3.6
+        'pytest-timeout<1.3.0',
+        'pytest-cov',
+        'nose',
```
Comment (kmaehashi) [inline setup.py]: * `nose` is no longer necessary to run Chainer tests.
* How about adding `attrs` restrictions as well? https://github.com/chainer/chainer-test/pull/529

Comment (niboshi) [reply setup.py]: How about separate PRs?
The purpose of this PR is not to change requirements.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: setup.py (hunk 1 above)
Comment (kmaehashi) [reply setup.py]: Ok.

**Comment by kmaehashi** [comment]
The Jenkins failure is not related to this PR and will be separately fixed.

**Pull request merged**